File: 405.convert.a.number.to.hexadecimal.cpp
class Solution {
public:
    string toHex(unsigned int num) {
        static const char hex[] = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f"};
        if (num == 0) return "0";
        string ret = "";
        for(; num; ret=hex[num&15] + ret, num = num >> 4);
        return ret;
    }
};

File: 405. Convert a Number to Hexadecimal.py
class Solution:
    def toHex(self, num: int) -> str:
        if num == 0:
            return "0"
        hexstr = "0123456789abcdef"
        if num < 0:
            num = (1<<32) + num
        ret = []
        while num > 0:
            x = num % 16
            ret.append(hexstr[x])
            num = num // 16
        ret.reverse()
        return "".join(ret)
        

